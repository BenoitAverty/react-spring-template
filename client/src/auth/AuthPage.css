.Login-form button {
  width: 40%;
  margin: auto;
  display: block;
}

.LoggedInWidget {
  text-align: center;
}
