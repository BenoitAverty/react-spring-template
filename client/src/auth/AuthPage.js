import React from 'react'
import {connect} from 'react-redux'

import {login, logout} from './authReducer'

import LoginForm from './LoginForm'
import LoggedInWidget from './LoggedInWidget'
import './AuthPage.css'

function AuthPage(props) {
  if (props.username === '') {
    return <LoginForm
      onSubmit={props.onSubmit}
      disabled={props.loginInProgress}
           />
  }
  else {
    return <LoggedInWidget
      username={props.username} onLogout={props.onLogout} />
  }
}

const mapStateToProps = ({ username, loginInProgress }) => ({
  username,
  loginInProgress
})
function mapDispatchToProps(dispatch) {
  return {
    onSubmit: credentials => dispatch(login(credentials)),
    onLogout: () => dispatch(logout())
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(AuthPage)
