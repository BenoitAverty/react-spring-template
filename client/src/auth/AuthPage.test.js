import React from 'react';

import AuthPage from './AuthPage';

it('Matches latest snapshot', () => {
  const element = render(<AuthPage />)

  expect(element).toMatchSnapshot()
});
