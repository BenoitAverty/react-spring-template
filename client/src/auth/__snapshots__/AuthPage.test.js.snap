exports[`test Matches latest snapshot 1`] = `
<div>
  <Form
    bsClass="form"
    className="Login-form"
    componentClass="form"
    horizontal={true}
    inline={false}>
    <FormGroup
      bsClass="form-group"
      controlId="username">
      <Col
        bsClass="col"
        componentClass="div"
        sm={2}>
        <ControlLabel
          bsClass="control-label"
          srOnly={false}>
          Username
        </ControlLabel>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        sm={10}>
        <FormControl
          bsClass="form-control"
          componentClass="input"
          onChange={[Function]}
          type="text"
          value="" />
      </Col>
    </FormGroup>
    <FormGroup
      bsClass="form-group"
      controlId="password">
      <Col
        bsClass="col"
        componentClass="div"
        sm={2}>
        <ControlLabel
          bsClass="control-label"
          srOnly={false}>
          Password
        </ControlLabel>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        sm={10}>
        <FormControl
          bsClass="form-control"
          componentClass="input"
          onChange={[Function]}
          type="text"
          value="" />
      </Col>
    </FormGroup>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
      type="submit">
      Login
    </Button>
  </Form>
</div>
`;
